include/LightGBM/tree.h

```diff
@@ -189,9 +189,21 @@ class Tree {
     for (int i = 0; i < num_leaves_ - 1; ++i) {
       leaf_value_[i] = MaybeRoundToZero(leaf_value_[i] * rate);
       internal_value_[i] = MaybeRoundToZero(internal_value_[i] * rate);
+      if (is_linear_) {
+        leaf_const_[i] = MaybeRoundToZero(leaf_const_[i] * rate);
```

Comment:
I don't think so. These values are not the same, and we need both because when we predict for a row containing nan feature values we can't use the linear model and so we predict `leaf_value_`.